src/environmentd/src/http.rs

```diff
@@ -426,6 +429,24 @@ impl InternalHttpServer {
     }
 }
 
+async fn internal_http_auth<B>(mut req: Request<B>, next: Next<B>) -> impl IntoResponse {
+    if let Some(user) = req.headers().get("x-materialize-user") {
+        if user == SUPPORT_USER_NAME {
```

Comment:
i think setting `x-materialize-user=mz_system` would error out. I assume you don't want that, but not sure

Reply:
Good point! Fixed.